geo/src/algorithm/concave_hull.rs

```diff
@@ -121,8 +121,10 @@ where
 {
     let h = max_dist + max_dist;
     let w = line.euclidean_length() + h;
+    let diagonal_squared = T::powi(h, 2) + T::powi(w, 2);
     let two = T::add(T::one(), T::one());
-    let search_dist = T::div(T::sqrt(T::powi(w, 2) + T::powi(h, 2)), two);
+    let four = T::add(two, two);
```

Comment:
I'm having a little trouble following this - does this method correspond to concaveman's [`findCandidate`](https://github.com/mapbox/concaveman/blob/54838e1c9eb416f4393438df7e78ac2b0ca7faa5/index.js#L94) method?  Or has our implementation diverged sufficiently that there's not really a corollary?

Reply:
> concaveman uses a rectangular envelope where the borders are a distance of max_dist from the line to look for points. Unfortunately, rstar uses a circular envelope

That's interesting - is that something inherent in the rstar implementation? Can you link to any doc or code example that highlights this difference?

Reply:
The infix notation usage sounds reasonable. I'll set that up. 

With regard to `findCandidate`, yeah, this is reasonably equivalent. The main difference is that `findCandidate` implements the traversal of the queue within itself while I offload that onto rstar.

With regard to where concaveman uses a rectangular envelope, I'm embarrassed to admit that it's not entirely clear from looking at the code again where that is. With regard to rstar, I use [`locate_within_distance`](https://docs.rs/rstar/0.8.2/rstar/struct.RTree.html#method.locate_within_distance) which references a radius r which I took to mean that it looks for points within a circle of radius r.  I'll have to dive into the concaveman code again.

Reply:
@michaelkirk From checking again, it seems I was wrong to think that concaveman used a rectangular envelope to search within. It seems to be using a circle just like rstar, I'm not entirely sure where I got the impression that it used a rectangular envelope. I've adjusted the calculation to be simpler.

Reply:
Thanks for double checking the envelop shape stuff @RestitutorOrbis!

Since we don't have many tests to debug edge cases, anything we can do to make our code look more like the reference implementation (assuming that's correct) would make review easier.

Since our implementation has diverged from concaveman it was hard to see, but it looks like we compute max distance as the `(line_length / concavity)^2` whereas concaveman has max distance as `(line_length^2 / concavity^2)`. Is that divergence intentional?

Reply:
That divergence was intentional on the basis that those are equivalent statements anyways (ex. `(x/y)^2=x^2/y^2`) and it would be easier to understand the function call if it was handing over `dist` instead of a stranger notion of `squared_dist`.

Reply:
🤣 apparently I wasn't ready to do 4th grade math yesterday AM.